src/RichTextEditor/components/index.js

```diff
@@ -20,14 +20,27 @@ const Element = (props) => {
     attributes, children, element, customElements
   } = props;
   const { type, data } = element;
+  const headings = [H1, H2, H3, H4, H5, H6];
+  let headingId;
+  if (headings.includes(type)) {
+    // https://github.com/thlorenz/anchor-markdown-header/blob/87e4cca4618271e363f4af9697df0f73f23b3d3f/anchor-markdown-header.js#L20
+    headingId = element.children[0].text.replace(/ /g,'-')
+    // escape codes
+    .replace(/%([abcdef]|\d){2,2}/ig, '')
+    // single chars that are removed
+    .replace(/[\/?!:\[\]`.,()*"';{}+=<>~\$|#@&–—]/g,'')
+    // CJK punctuations that are removed
+    .replace(/[。？！，、；：“”【】（）〔〕［］﹃﹄“ ”‘’﹁﹂—…－～《》〈〉「」]/g, '')
+    ;
+  }
```

Comment:
Could we make the `headingId` creation a utility function and declare `headingId` in a ternary?